_posts/2020-04-13-deprecation-warnings-rails-guide.markdown

```diff
@@ -0,0 +1,107 @@
+---
+layout: post
+title: "The Complete Guide for Deprecation Warnings in Rails"
+date: 2020-04-13 12:00:00
+categories: ["rails", "upgrades"]
+author: luciano
+---
+
+Deprecation warnings are a common thing in the software world. They are warnings that notify the developer that a specific feature (e.g. a method) will be removed soon (usually in the next [Minor](https://semver.org/) version) and should be replaced with something else.
+Features are deprecated rather than immediately removed, to provide backward compatibility (a solution that works in both the current and the future version), and to give programmers time to implement the code in a way that follows the new standard.
```

Comment:
```suggestion
Features are deprecated rather than immediately removed, in order to provide backward compatibility (a solution that works in both the current and the future version), and to give programmers time to implement the code in a way that follows the new standard.
```